python/GangaDirac/Lib/RTHandlers/DiracRTHUtils.py

```diff
@@ -87,48 +96,42 @@ def dirac_outputfile_jdl(output_files, empty_SE_check):
 
 
 def dirac_inputdata(app, hasOtherInputData=False):
-    """ Construct the JDL compoenent which requests the inputdata for a job """
-    job = stripProxy(app).getJobObject()
+    """ Construct the JDL compoenent which requests the inputdata for a job
+    Args:
+        app (IApplication): app which belongs to the job of interest
+        hasOtherInputData (bool): This is used to stop BannedSites being added to the JDL structure through backend.settings
+    """
+    job = app.getJobObject()
     input_data = None
     parametricinput_data = None
 
     inputLFNs = []
 
-    if hasattr(job.inputdata, 'getLFNs'):
-        inputLFNs = job.inputdata.getLFNs()
-
-    if job.master:
-        logger.debug("job.master.inputdata: %s " % str(job.master.inputdata))
-    logger.debug("job.inputdata: %s" % str(job.inputdata))
-    if hasattr(job.inputdata, 'getLFNs'):
-        logger.debug("getLFNs(): %s" % job.inputdata.getLFNs())
-
-    has_input_DiracFile = False
-    for this_file in job.inputfiles:
-        if isType(this_file, DiracFile):
-            has_input_DiracFile = True
-            break
-    if job.master and not has_input_DiracFile:
-        for this_file in job.master.inputfiles:
-            if isType(this_file, DiracFile):
-                has_input_DiracFile = True
-                break
+    input_data = None
+    parametricinput_data = None
 
-    if len(inputLFNs) > 0:
-        # master job with a splitter reaching prepare, hence bulk submit
-        if not job.master and job.splitter:
-            parametricinput_data = dirac_parametric_split(app)
-            if parametricinput_data is not None and len(parametricinput_data) > getConfig('DIRAC')['MaxDiracBulkJobs']:
-                raise BackendError('Dirac', 'Number of bulk submission jobs \'%s\' exceeds the maximum allowed \'%s\' if more are needed please modify your config. Note there is a hard limit in Dirac of currently 1000.' % (
-                    len(parametricinput_data), getConfig('DIRAC')['MaxDiracBulkJobs']))
+    if not job.inputdata and not job.master.inputdata:
+        return input_data, parametricinput_data
+
+    wanted_job = job
+    if not job.inputdata and job.master.inputdata:
+        wanted_job = job.master
+
+    inputLFNs = ['LFN:'+this_file.lfn for this_file in wanted_job.inputdata if isType(this_file, DiracFile)]
```

Comment:
No, No, No and doing so breaks things for some awful reason.
I think this is because the Dataset is returning a proxied object from a dataset